tests/tel2puml/otel_to_pv/data_holders/sql_data_holder/test_sql_dataholder.py

```diff
@@ -703,3 +695,168 @@ def test_find_unique_graphs(
+        )
+    )
+
+    num_jobs = 10
+    events_per_job = 10000  # Total of 100,000 events
+
+    with large_sql_data_holder:
+        for job_index in range(num_jobs):
+            job_name = f"job_{job_index}"
+            job_id = f"job_id_{job_index}"
+            for event_index in range(events_per_job):
+                otel_event = OTelEvent(
+                    job_name=job_name,
+                    job_id=job_id,
+                    event_type=f"event_type_{event_index}",
+                    event_id=f"event_{job_index}_{event_index}",
+                    start_timestamp=1723544132228288000 + event_index,
+                    end_timestamp=1723544132228288000 + event_index + 10,
+                    application_name="test_application",
+                    parent_event_id=None,
+                    child_event_ids=[],
+                )
+                large_sql_data_holder.save_data(otel_event)
+
+    # Stream data and verify counts
+    total_events_streamed = 0
+    with large_sql_data_holder:
+        result = large_sql_data_holder.stream_data()
+        for job_name, otel_event_gen in result:
+            event_count = 0
+            for otel_event in otel_event_gen:
+                event_count += 1
+            assert event_count == events_per_job
+            total_events_streamed += event_count
+
+    assert total_events_streamed == num_jobs * events_per_job
+
+
+def test_stream_job_name_batches(
+    sql_data_holder_with_multiple_otel_job_names: SQLDataHolder,
+) -> None:
+    """Test stream_job_name_batches method."""
+    # Test 1: Stream all data
+    sql_data_holder = sql_data_holder_with_multiple_otel_job_names
+    with sql_data_holder.session as session:
+        otel_event_gen = sql_data_holder.stream_job_name_batches(session)
+        all_events = list(otel_event_gen)
+
+    assert len(all_events) == 20
+
+    # Test 2: Stream data, filtered by job name
+    with sql_data_holder.session as session:
+        otel_event_gen = sql_data_holder.stream_job_name_batches(
+            session, filter_job_names={"test_name"}
+        )
+        all_events = list(otel_event_gen)
+
+    assert all(otel_event.job_name == "test_name" for otel_event in all_events)
+    assert len(all_events) == 10
+
+    # Test 3: Stream data, filtered using map
+    with sql_data_holder.session as session:
+        otel_event_gen = sql_data_holder.stream_job_name_batches(
+            session,
+            job_name_to_job_ids_map={"test_name": {"test_id_0", "test_id_1"}},
+        )
+        all_events = list(otel_event_gen)
+
+    assert len(all_events) == 4
+    for otel_event in all_events:
+        assert otel_event.job_name == "test_name"
+        assert otel_event.job_id in {
+            "test_id_0",
+            "test_id_1",
+        }
+
+    # Test 4: Stream data, filtering with name and map
+    filter_job_names = {"test_name"}
+    job_name_to_job_ids_map = {"test_name": {"test_id_0", "test_id_1"}}
```

Comment:
should have test_name_1 in the dictionary as I can't tell that the filter is working with the map here